shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp

```diff
@@ -85,6 +109,24 @@ namespace shared_model {
       // stub hash
       const Lazy<crypto::Hash> hash_{
           [this] { return crypto::Hash(this->proto_->tx_hash()); }};
+
+      /**
+       * @return priority of this transaction response; transaction response can
+       * only be replaced with one with higher priority
+       */
+      int priority() const noexcept {
+        return iroha::visit_in_place(
+            *variant_,
+            [](const StatelessValidTxResponse &) { return 1; },
+            [](const MstPendingResponse &) { return 2; },
+            [](const EnoughSignaturesCollectedResponse &) { return 3; },
+            [](const StatefulValidTxResponse &) { return 4; },
+            [](const CommittedTxResponse &) { return 5; },
+            [](const StatelessFailedTxResponse &) { return 6; },
+            [](const StatefulFailedTxResponse &) { return 7; },
+            [](const MstExpiredResponse &) { return 8; },
+            [](const NotReceivedTxResponse &) { return 9; });
```

Comment:
Currently, statuses are misleading. For example, why MstExpired is greater than committed?

I suggest reworking priorities with final status priority, which means maximal priority and reflects the impossibility of achieving another status for the transaction. Also, take care of intermediate statuses.

If there will be some troubles, we may arrange an offline meeting.